src/Node/Node.php

```diff
@@ -207,18 +207,12 @@ public function removeNode($name)
         unset($this->nodes[$name]);
     }
 
-    /**
-     * @return int
-     */
-    public function count()
+    public function count(): int
```

Comment:
This is a breaking change, thus I think we have to use `#[\ReturnTypeWillChange]` here.

Reply:
I see. I'm not entirely sure why (and what will break). 
I just corrected it to what you suggested, though. 

Should I do the same for getIterator?

... changing  

```
    public function getIterator(): \Traversable
```

to

```
    #[\ReturnTypeWillChange]
    public function getIterator()
```

?

Reply:
The problem is, that adding a return type in a parent class forces child classes to also add them. So if anyone extends this class, it'll break until they add the return type there, too.

Reply:
Ahh! I see. 

I just committed the change to `getIterator()` as well. 

Thanks a lot. <3

Reply:
Here's some more information: https://www.php.net/manual/en/language.oop5.variance.php

> Covariance allows a child's method to return a more specific type than the return type of its parent's method.

But not the other way around! So, child class return types may be *more* specific, but not *less* specific.